Block/Catalog/Product/View/PaypalExpress.php

```diff
@@ -0,0 +1,139 @@
+ *
+ * @copyright Copyright (c) Buckaroo B.V.
+ * @license   https://tldrlegal.com/license/mit-license
+ */
+
+namespace Buckaroo\Magento2\Block\Catalog\Product\View;
+
+use Magento\Framework\Encryption\Encryptor;
+use Magento\Framework\View\Element\Template;
+use Buckaroo\Magento2\Model\ConfigProvider\Account;
+use Magento\Framework\View\Element\Template\Context;
+use Buckaroo\Magento2\Model\ConfigProvider\Method\Paypal;
+
+class PaypalExpress extends Template
+{
+    /**
+     * @var \Buckaroo\Magento2\Model\ConfigProvider\Account
+     */
+    protected $configProviderAccount;
+
+    /**
+     * @var \Magento\Framework\Encryption\Encryptor
+     */
+    protected $encryptor;
+
+    /**
+     * @var \Buckaroo\Magento2\Model\ConfigProvider\Method\Paypal
+     */
+    protected $paypalConfig;
+
+    public function __construct(
+        Context $context,
+        Account $configProviderAccount,
+        Encryptor $encryptor,
+        Paypal $paypalConfig,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->configProviderAccount = $configProviderAccount;
+        $this->encryptor = $encryptor;
+        $this->paypalConfig = $paypalConfig;
+    }
+    public function canShowProductButton()
+    {
+        return $this->canShowButtonForPage('Product');
+    }
+    public function canShowCartButton()
+    {
+        return $this->canShowButtonForPage('Cart');
+    }
+    /**
+     * Get all data required
+     *
+     * @return array
+     */
+    public function getConfig()
+    {
+        return [
+            'currency' => $this->getCurrency(),
+            'buckarooWebsiteKey' => $this->getWebsiteKey(),
+            'paypalMerchantId' => $this->getMerchantId(),
+        ];
+    }
+    /**
+     * Get buckaroo website key
+     *
+     * @return string
+     */
+    protected function getWebsiteKey()
+    {
+        return $this->encryptor->decrypt(
+            $this->configProviderAccount->getMerchantKey(
+                $this->_storeManager->getStore()
+            )
+        );
+    }
+    /**
+     * Get shop currency
+     *
+     * @return string
+     */
+    protected function getCurrency()
+    {
+        return $this->_storeManager
+            ->getStore()
+            ->getCurrentCurrency()
+            ->getCode();
+    }
+    /**
+     * Test if button was enabled for this page 
+     *
+     * @param string $page
+     *
+     * @return boolean
+     */
+    protected function canShowButtonForPage($page)
+    {
+        $buttons = $this->paypalConfig->getExpressButtons(
+            $this->_storeManager
+                ->getStore()
+                ->getId()
+        );
+        if ($buttons === null) {
+            return false;
+        }
+
+        $pages = explode(",", $buttons);
+        return in_array($page, $pages);
+    }
+    /**
+     * Get merchant id
+     *
+     * @return string|null
+     */
+    protected function getMerchantId()
```

Comment:
same - part of configprovider.

Reply:
this is only one call to the provider function, if i remove this, i will need to call the provider directly in the phtml